src/viam/sdk/common/proto_type.cpp

```diff
@@ -170,16 +172,21 @@ bool operator==(const ProtoType& lhs, const ProtoType& rhs) {
                 return false;
             }
             // NOLINTNEXTLINE(misc-no-recursion)
-            auto pred = [](auto lhs_map, auto rhs_map) {
-                return lhs_map.first == rhs_map.first && *lhs_map.second == *rhs_map.second;
-            };
-
-            return std::equal(lhs_map->begin(), lhs_map->end(), rhs_map->begin(), pred);
+            for (const auto& kv : *lhs_map) {
```

Comment:
This map comparison was failing on larger maps because the iteration order seemed inconsistent. This works now.